src/controls/LevelingSensor/QskLevelingSensorUtility_test.cpp

```diff
@@ -0,0 +1,190 @@
+#include <cstdlib>
+#include <type_traits>
+#include <iostream>
+#include <string>
+
+#include <QSGNode>
+
+#include "QskLevelingSensorUtility.h"
+
+#define ASSERT_TRUE(expr) if(!(expr)) { std::cerr << "ASSERT_TRUE(!" << #expr << "):" << __LINE__ << '\n'; return false; }
```

Comment:
I guess you could just use QVERIFY from QtTest for this, which also prints the line IIRC